fw/src/mppt/global/global_algorithms.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file global_algorithms.hpp
+ * @author Praneel Murali (praneelm@utexas.edu)
+ * @brief Default Global MPPT algorithm driver.
+ * @version 0.2.0
+ * @date 2024-03-18
+ *
+ * @copyright Copyright (c) 2024
+ *
+ */
+
+#pragma once
+#include "../mppt.hpp"
+
+// Global Algorithm Selection
+#define GLOBAL_ALGO 0
+// 0: Default (Local Algorithm Only)
+// 1: Voltage Sweep
+// 2: Trapezoidal Sum Optimization
+// 3: Simulated Annealing
+// 4: Firefly Algorithm
+
+// Local Algorithm Selection
+#define LOCAL_ALGO 0
+// 0: Perturb and Observe (PandO)
+// 1: Incremental Conductance (IC)
+// 2: Fuzzy Logic
+// 3: Feedback Control (FC)
+
+#if LOCAL_ALGO == 0
+#include "../local/pando.hpp"
+#elif LOCAL_ALGO == 1
+#include "../local/ic.hpp"
+#elif LOCAL_ALGO == 2
+#include "../local/fuzzy.hpp"
+#elif LOCAL_ALGO == 3
+#include "../local/fc.hpp"
+#endif
+
+class DefaultGlobal final : public MPPT {
+public:
+    DefaultGlobal(void) : MPPT() {
+        #if LOCAL_ALGO == 0
+        localMPPT = new PandO();
```

Comment:
not particularly enthused about the use of new for embedded systems context
This keyword implies that the new data structure is created on heap, which is:
1. very expensive and slower than static memory
2. not even necessarily supported (see TM4C), so not platform independent

It would be preferable to pass in a reference to an already instantiated local algorithm (which could be a MPPT instance with or without distinction from global algorithms).

That way, you also don't care about the checks and just run it as is with the standard interface provided in mppt.hpp. 

Ideally any global algorithm implementation will work without any special code for any particular local algorithm: you could theoretically just swap one out in runtime and keep running without issues.